pep-0544.txt

```diff
@@ -771,17 +786,21 @@ Implementation details
 
 The runtime implementation could be done in pure Python without any
 effects on the core interpreter and standard library except in the
-``typing`` module:
+``typing`` module, and a minor update to ``collections.abc``:
 
 * Define class ``typing.Protocol`` similar to ``typing.Generic``.
 * Implement metaclass functionality to detect whether a class is
-  a protocol or not. Add a class attribute ``__protocol__ = True``
+  a protocol or not. Add a class attribute ``_is_protocol = True``
   if that is the case. Verify that a protocol class only has protocol
   base classes in the MRO (except for object).
-* Implement ``@runtime`` that adds all attributes to ``__subclasshook__()``.
+* Implement ``@runtime`` that allows ``__subclasshook__()`` performing
+  structural instance and subclass checks as in ``collections.abc`` classes.
 * All structural subtyping checks will be performed by static type checkers,
   such as ``mypy`` [mypy]_. No additional support for protocol validation will
   be provided at runtime.
+* Classes ``Mapping``, ``MutableMapping``, ``Sequence``, and
+  ``MutableSequence`` in ``collections.abc`` module will support structural
+  instance and subclass checks (like e.g. ``collections.abc.Iterable``).
```

Comment:
This is actually an optional change. I have a small branch in my CPython fork for this. I propose this for two reasons:
* This makes ``typing.Mapping`` and ``collections.abc.Mapping`` interchangeable/consistent. It would be strange/unexpected if all protocols in ``typing`` and ``collections.abc`` behave the same way _except_ for these four.
* This makes implementation in ``typing`` cleaner.

A possible strategy would be to use ``if sys.version_info >= ...`` in ``typing`` implementation, so that the behaviour of ``typing`` protocols is the same in all versions, and we can safely make the ``collections.abc`` change only in 3.7.